src/cvl/custom_validation/custom_validation.go

```diff
@@ -116,3 +118,66 @@ func (t *CustomValidation) ValidateAclRuleIPAddress(
 	return CVLErrorInfo{ErrCode: CVL_SUCCESS}
 }
 
+//Path : /sonic-sflow/SFLOW/SFLOW_LIST/agent_id
+//Purpose: Check correct for correct agent_id
+//vc : Custom Validation Context
+//Returns -  CVL Error object
+func (t *CustomValidation) ValidateSflowAgentId(
+	vc *CustValidationCtxt) CVLErrorInfo {
+
+	log.Info("ValidateSflowAgentId operation: ", vc.CurCfg.VOp)
+	if (vc.CurCfg.VOp == OP_DELETE) {
+		return CVLErrorInfo{ErrCode: CVL_SUCCESS}
+	}
+
+	log.Info("ValidateSflowAgentId YNodeVal: ", vc.YNodeVal)
+
```

Comment:
Please add fix to "sflow enable failed  with default settings failed" below. If intending to merge later, please disregard.
      /*  allow empty or deleted agent_id */
      if vc.YNodeVal == "" {
          return CVLErrorInfo{ErrCode: CVL_SUCCESS}
      }